Common/Interfaces/IAlgorithm.cs

```diff
@@ -328,6 +328,11 @@ IFutureChainProvider FutureChainProvider
             get;
         }
 
+        /// <summary>
+        /// Gets the market hours database in use by this algorithm
+        /// </summary>
+        MarketHoursDatabase MarketHoursDatabase { get; }
```

Comment:
I'm reluctant about this. In theory, this is a great idea, but I have concerns on the direction of the dependency. Currently QCAlgorithm is hard-coded to fetch `FromDataFolder()` (as is the rest of the application). I love the intent behind this, making things instance based, but if we're going to make accesses to the market hours db instance based we should tackle that in a separate PR and update the solution to follow that new pattern.